### PR TITLE
REF: simplify BlockManager.quantile

### PR DESCRIPTION
**Pull request by jbrockmendel**
Handle the scalar-qs case early on in DataFrame.quantile, hugely simplifies BlockManager.quantile and to a lesser extent Block.quantile

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -2185,6 +2176,7 @@ def fillna(
         )
 
     def quantile(self, qs, interpolation="linear", axis: int = 0) -> Block:
+        assert axis == 1  # only ever called this way
```
Comment (jreback) [inline pandas/core/internals/blocks.py:2179]: prob should just change the default? also can you type qs now / assert it?

Comment (jreback) [reply pandas/core/internals/blocks.py:2179]: i see you are doing that below already.

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -470,73 +467,26 @@ def quantile(
...
-                ax = Float64Index(qs)
-            else:
-                ax = axes[0]
-            return ax
+        qs_axe = Float64Index(qs)
+        new_axes = list(self.axes)
+        new_axes[1] = qs_axe
 
-        axes, blocks = [], []
+        blocks = []
```
Comment (jreback) [inline pandas/core/internals/managers.py:477]: can you make a list comprehension

**Review by jreback — CHANGES_REQUESTED** [review]
looks good some comments

File: pandas/core/internals/blocks.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:2179]: > type qs now

tried annotating and got unhelpful mypy complaints

File: pandas/core/internals/managers.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:477]: sure

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**